### ISSUE TITLE
New buzzwords The Artificial Intelligence Hardware and Software Dilemma

### ISSUE THREAD

**Issue by postmillenial** [postmillenial/reading-list#884]
The Artificial Intelligence Hardware and Software Dilemma<br>
http://ift.tt/2gwyWdU<br><img width="640" height="439" src="http://ift.tt/2yDcVkx" class="webfeedsFeaturedVisual wp-post-image wp-stateless-item" style="display: block; margin: auto; margin-bottom: 20px;" data-image-size="large" data-stateless-media-bucket="cdn.thenewstack.io" data-stateless-media-name="media/2017/10/c38ef166-neurala.jpg"><a href="http://ift.tt/2vprFQ6" class="clearfix infoBlock" target="_blank"></a>
<div class="infoBlockImageBlock"><img class="infoBlockImage" src="http://ift.tt/2yDcXc9" style="border-radius: 50%;"></div>
<div class="infoBlockTextBlock">
<div class="infoBlockTitle">Massimiliano Versace</div>
<div class="infoBlockText">Massimiliano Versace is the co-founder and CEO of Neurala Inc. and the company visionary. After his pioneering research in brain-inspired computing and deep networks, he continues to inspire and lead the world of autonomous robotics. He has spoken at a number of events and venues, including TedX, NASA, Los Alamos National Labs, Air Force Research Labs, HP, iRobot, Samsung, LG, Qualcomm, Ericsson, BAE Systems, Mitsubishi, ABB and Accenture. In addition, his work has been featured in IEEE Spectrum, New Scientist, Geek Magazine, CNN and MSNBC. Versace is a Fulbright scholar, has authored dozens of academic publications and holds several patents. He also holds two Ph.D.s: in Experimental Psychology from the University of Trieste and in Cognitive and Neural Systems from Boston University.</div>
</div>
<p>Artificial intelligence: everybody is talking about it, and the as-of-yet unrealized possibilities of the technology are fueling a renaissance in the hardware and software industry. Hardware and software companies &mdash; including Intel, NVidia, Google, IBM, Microsoft, Facebook, Qualcomm, ARM and many others &mdash; are racing to build the next AI hardware platform or fighting to maintain their lead.</p>
<p>AI, and deep learning (a sub-field of neural networks) in particular is an inherently non-<a href="http://ift.tt/2guDDoI" class="ext-link" rel="external nofollow">Von Neumann process</a>, and the prospect of having a processor more closely tailored to the specific needs of neural networks is appealing.</p>
<p>But, I like to think before acting, especially before diving into a potentially very expensive hardware project. And I would like to follow the very same paradigm that deep learning successfully uses to compute &mdash;&nbsp;<em>learn from past examples&nbsp;</em>&mdash; to determine the best path forward when considering this important question:</p>
<p>Should the AI industry build a specialized deep learning chip, and, if so, what should it look like?</p>
<p>Let&rsquo;s first look at the past to predict the future.</p>
<h3>The World Is Administered by the Wise but Progresses Because of the Crazy</h3>
<p>The U.S.&nbsp;<a href="https://www.darpa.mil/" class="ext-link" rel="external nofollow">Defense Advanced Research Projects Agency</a> (DARPA) is an agency created in 1958 as the Advanced Research Projects Agency (ARPA) by U.S. President Dwight Eisenhower with the goal of the developing &ldquo;crazy innovative&rdquo; technologies for use by the military. The agencies played a vital role in the development of defense (and non-defense) technology, such as&nbsp;<a href="http://ift.tt/fVNWcB" data-saferedirecturl="http://ift.tt/2gwzyQK" class="ext-link" rel="external nofollow">ARPANET</a>, the basis for what would become the internet; GPS technology; and many others.</p>
<p>In 2008, DARPA launched the Systems of Neuromorphic Adaptive Plastic Scalable Electronics (<a href="http://ift.tt/2yDcZRj" data-saferedirecturl="http://ift.tt/2gvlmYq" class="ext-link" rel="external nofollow">SyNAPSE</a>). The goal was to create a new breed of non-traditional brain-inspired processors that could perform perception and behavioral tasks while simultaneously achieving significant energy savings.</p>
<blockquote>
<p>Biology makes no distinction between memory and computation.</p>
</blockquote>
<p>The program, which originally involved my lab at Boston University and researchers at Hewlett-Packard, HRL Laboratories, and IBM, had the goal of achieving energy efficiency by intelligently locating data and computation on the chip, alleviating the need to move data over large distances. The idea was to do this from a single synapse and scale up to a whole brain (see figure).</p>
<p><a href="http://ift.tt/2yEzi91" class="ext-image" rel="external nofollow"><img class="aligncenter size-full wp-image-3271995" src="http://ift.tt/2yEzi91" width="1060" height="582" data-id="3271995"></a></p>
<p>Some versions of the chip employed sophisticated&nbsp;<a href="http://ift.tt/1nb7t7q" data-saferedirecturl="http://ift.tt/2gvU30g" class="ext-link" rel="external nofollow">memristive&nbsp;</a>device technologies to implement dense connectivity between neurons and asynchronous computation &mdash; processing and transmitting data only as required, similar to how neurons process and communicate information.</p>
<h2>Why SyNAPSE?</h2>
<p>SyNAPSE was a complex, multi-faceted project, but it was rooted in two fundamental problems. First, traditional algorithms perform poorly in the complex, real-world environments in which biological agents thrive. Biological computation, in contrast, is highly distributed and deeply data-intensive. Second, traditional microprocessors are extremely inefficient at executing highly distributed, data-intensive algorithms. SyNAPSE sought both to advance the state-of-the-art in biological algorithms and to develop a new generation of nanotechnology necessary for the efficient implementation of those algorithms.</p>
<p><a href="http://ift.tt/2yDd1bT" class="ext-image" rel="external nofollow"><img class="size-full wp-image-3271996 alignright" src="http://ift.tt/2yDd1bT" width="300" height="228" data-id="3271996"></a></p>
<p>Biology makes no distinction between memory and computation. Virtually every synapse of every neuron simultaneously stores information and uses this information to compute. Standard computers, in contrast, separate memory and processing into two nice, neat boxes. Biological computation assumes these boxes are the same thing. Understanding why this assumption is such a problem requires stepping back to the core design principles of digital computers.</p>
<p>The vast majority of current-generation computing devices are based on the&nbsp;<a href="http://ift.tt/hZiEHW" data-saferedirecturl="http://ift.tt/2yDd3R3" class="ext-link" rel="external nofollow">Von Neumann architecture</a>. This core architecture is wonderfully generic and multi-purpose, attributes which enabled the information age. Von Neumann architecture comes with a deep, fundamental limit, however. A Von Neumann processor can execute an arbitrary sequence of instructions on arbitrary data, enabling re-programmability, but the instructions and data must flow over a limited capacity bus, connecting the processor and main memory. Thus, the processor cannot execute a program faster than it can fetch instructions and data from memory. This limit is known as the &ldquo;Von Neumann bottleneck.&rdquo;</p>
<p>Multi-core and massively multi-core architectures are partial solutions but still fit within the same general theme. Extra transistors are swapped for higher performance. Rather than relying on automatic mechanisms alone, though, multi-core chips give programmers much more direct control of the hardware. This works beautifully for many classes of algorithms, but not all and certainly not for data-intensive bus-limited ones.</p>
<p>Unfortunately, the exponential transistor density growth curve cannot continue forever without hitting basic physical limits, and these transistors will not serve the purpose of enabling brains to run efficiently in silicon. Enter the DARPA SyNAPSE program.</p>
<h2>The Outcome of SyNAPSE</h2>
<p>In 2014, after about six years of effort, one of the groups &mdash; IBM in San Jose, California &mdash; announced the &ldquo;birth&rdquo; of its SyNAPSE chip, loaded with more than five billion transistors and more than 250 million synapses (programmable logic points). The chip was built on Samsung Foundry&rsquo;s 28nm process technology, and its power consumption was less than 100 milliwatts during operation. Despite this still being orders of magnitude fewer than the actual count of brain synapses, it was an important step toward building more brain-like chips.</p>
<p><a href="http://ift.tt/2gwzCzY" class="ext-image" rel="external nofollow"><img class="aligncenter size-full wp-image-3271997" src="http://ift.tt/2gwzCzY" width="619" height="316" data-id="3271997"></a></p>
<p>So, after DARPA poured tens of millions of dollars into SyNAPSE, where are neuromorphic&mdash;&ldquo;brain-inspired&rdquo;&mdash;chips today? Can I find them at Best Buy, as I can find the other processors used today to run AI?</p>
<p>They are nowhere to be found.</p>
<p>So, what can be found? Which hardware can AI run on?</p>
<h2>CPU, GPU, VPU, TPU, [whatever you want]PU</h2>
<p>While you can&rsquo;t buy a fancy SyNAPSE processor to run your AI, you have surely heard of &mdash; and maybe have even bought &mdash; CPUs or GPUs (graphics processors). <a href="http://ift.tt/2vnbyod" class="local-link">GPU-accelerated computing</a> leverages GPUs to reduce the time in computing complex applications: it offloads compute-intensive portions of the application to the GPU, while the remainder of the code still runs on the CPU. An intuitive way to understand why this is faster is to look into how CPUs and GPUs compute: while CPUs are designed to have a few cores optimized for sequential serial processing, GPUs have been built by design as massively parallel processors consisting of thousands of smaller cores which are best at handling multiple tasks simultaneously.</p>
<p><a href="http://ift.tt/2kScw6v" class="ext-link" rel="external nofollow">Movidius</a>, now part of Intel (as the FPGA company <a href="https://www.altera.com/" class="ext-link" rel="external nofollow">Altera</a> and the AI hardware company <a href="http://ift.tt/2tdC3vz" class="ext-link" rel="external nofollow">Nervana</a>), built and successfully commercialized a Visual Processing Unit (VPU). VPUs differ from GPUs as they were designed from the ground up to allow mobile devices to process intensive computer vision algorithms (whereas GPUs were adapted from processing graphics to machine vision).</p>
<p>Aside from <a href="http://ift.tt/g9uipl" class="ext-link" rel="external nofollow">NVidia</a> (GPUs) and Intel (CPUs and VPUs), <a href="http://bit.ly/2x5VLZD" class="ext-link" rel="external nofollow">Google</a> has joined the battle with the <a href="http://ift.tt/2qBeiNo" class="ext-link" rel="external nofollow">Tensor Processing Unit</a> (TPU). This is a custom ASIC specifically built for Google&rsquo;s <a href="http://ift.tt/2oPBp2x" class="local-link">TensorFlow</a> framework. The energy/speed savings are achieved by being more lenient to compute precision losses: by allowing the chip to be more tolerant of reduced computational precision, the TPU requires fewer transistors per operation. Since neural networks can tolerate such a precision loss by gracefully degrading performance (rather than &ldquo;STOP WORKING&rdquo;), it&rsquo;s a processor that simply works.</p>
<p>How about <a href="http://ift.tt/1Ber7Jm" class="ext-link" rel="external nofollow">Qualcomm</a>? The company has talked up, and then apparently abandoned, its <a href="https://www.zeroth.ai/" class="ext-link" rel="external nofollow">Zeroth AI</a> software platform for mobile computing, but it&rsquo;s hard to get a clear sense of Qualcomm&rsquo;s efforts in this area, although recent efforts on <a href="http://ift.tt/WTSLfj" class="ext-link" rel="external nofollow">Snapdragon</a> seem to indicate a resurgence of interest.</p>
<p>Finally, we should not forget <a href="https://www.arm.com/" class="ext-link" rel="external nofollow">ARM</a>: the chip-licensing firm &mdash; whose claim to fame is low-power processing &mdash; is heavily researching and investing in AI. And, to be honest, we can run <a href="http://ift.tt/2vprFQ6" class="ext-link" rel="external nofollow">Neurala</a>&rsquo;s algorithms pretty well on those tiny and inexpensive chips already!</p>
<p>So, there is plenty of hardware available to run your AI. What is the common thread of this hardware, and is it the time for the hardware industry to jump into another &ldquo;specialized AI chip?&rdquo;</p>
<h2>3 Lessons Learned from Brain-Inspired Hardware</h2>
<p>Programmability, programmability, programmability.</p>
<p>The success of NVidia is only partially due to performance. There are alternatives to NVidia processors that are either faster, leaner in power requirements or both. But NVidia&rsquo;s sweet spot is a judicious intersection of performance, price and ease of use. The company has done a very good job of making the development, testing and fielding of neural network algorithms seamless.</p>
<p>So, why is programmability so important for AI?</p>
<p>Because AI is still a very young and dynamic field. Neural networks, deep learning and reinforcement learning: these are all research-intensive enterprises. In research, flexibility and ease of use are key to shortening the time for a research cycle. A new paradigm or a new variation in an AI algorithm may modify the underlying mathematical and computational assumptions enough to make any specialized hardware hopelessly obsolete.</p>
<p>I truly believe that, in the best interest of advancing AI, hardware and software should be somehow factorized, as to allow maximum flexibility to algorithm developers, in particular at a time when AI is quickly innovating and finding its path.</p>
<p>I believe that the hardware company that understands this trend and provides flexibility as the first and foremost feature will succeed in capturing the heart &mdash; and brains &mdash; of the field.</p>
<p class="attribution"><a href="http://bit.ly/2x5VLZD" class="ext-link" rel="external nofollow">Google</a> is a sponsor of The New Stack.</p>
<p class="attribution">Images: Neurala, DARPA.</p>
<p>The post <a rel="nofollow" href="http://ift.tt/2gwyWdU">The Artificial Intelligence Hardware and Software Dilemma</a> appeared first on <a rel="nofollow" href="https://thenewstack.io">The New Stack</a>.</p>
<br><br>
via Blog &ndash; The New Stack https://thenewstack.io<br>
October 20, 2017 at 06:00AM
